payments-core/src/test/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowControllerTest.kt

```diff
@@ -409,6 +411,46 @@ internal class DefaultFlowControllerTest {
         flowController.confirm()
     }
 
+    @Test
+    fun `confirmPaymentSelection() with generic payment method should start paymentController`() =
```

Comment:
The test uses a new card and not a `GenericPaymentMethod`, so the name is incorrect. Can you add one test case for each?